integrations/tensorflow/iree_tf_compiler/TFL/LowerGlobalTensors.cpp

```diff
@@ -112,85 +110,72 @@ class LowerGlobalTensorsPass
         loc = std::get<1>(*locIt);
       }
 
-      std::string flowSymName = "__iree_flow_" + name.str();
-
       // TODO(suderman): Determine the global type based on all store
       // operations.
-      auto global = builder.create<iree_compiler::IREE::Input::GlobalOp>(
-          loc, flowSymName, /*is_mutable=*/true, attribute.getType(),
-          attribute);
+      std::string symName = "__global_" + name.str();
+      auto global = builder.create<mlir::ml_program::GlobalOp>(
+          loc, symName, attribute.getType(), /*is_mutable=*/true, attribute,
+          nullptr);
       global.setPrivate();
-      sharedNameToFlowName[name] = std::move(flowSymName);
-    }
 
-    // Replace handles with global addresses.
-    for (auto handle : handleOps) {
-      auto name = handle.shared_name();
-      auto flowName = sharedNameToFlowName[name];
-      auto constIt = sharedNameToConstant.find(name);
-      if (constIt == sharedNameToConstant.end()) continue;
-
-      auto attribute = std::get<1>(*constIt);
-
-      builder.setInsertionPoint(handle);
-      auto address =
-          builder.create<iree_compiler::IREE::Input::GlobalAddressOp>(
-              handle.getLoc(),
-              iree_compiler::IREE::Input::PtrType::get(context,
-                                                       attribute.getType()),
-              SymbolRefAttr::get(builder.getContext(), flowName));
-      handle.getResult().replaceAllUsesWith(address.getResult());
-      handle.erase();
+      symbolRefMap[name] = global;
     }
 
     // Replace the assign ops with a global store operation.
     for (auto assign : assignOps) {
-      auto address = dyn_cast<iree_compiler::IREE::Input::GlobalAddressOp>(
+      auto handle = dyn_cast<mlir::TFL::VarHandleOp>(
           assign.resource_id().getDefiningOp());
-      if (!address) continue;
+      if (!handle) continue;
 
-      builder.setInsertionPoint(assign);
       Value value = assign.value();
-      Type storageType = address.getType()
-                             .cast<iree_compiler::IREE::Input::PtrType>()
-                             .getTargetType();
-      if (storageType != value.getType()) {
+      auto globalOpIt = symbolRefMap.find(handle.shared_name());
+      if (globalOpIt == symbolRefMap.end()) continue;
```

Comment:
Should we warn in this case? This seems like a symbol that was referenced but wasnt' a global

Reply:
Added a warning statement that there was no corresponding GlobalOp.